src/upload.ts

```diff
@@ -1,31 +1,59 @@
+import { AxiosResponse } from "axios";
+import { Buffer } from "buffer";
+import { Upload } from "tus-js-client";
+
 import { getFileMimeType } from "./utils/file";
 import { BaseCustomOptions, defaultBaseOptions } from "./utils/options";
 import { formatSkylink } from "./skylink/format";
-import { SkynetClient } from "./client";
-import { AxiosResponse } from "axios";
+import { buildRequestHeaders, buildRequestUrl, SkynetClient } from "./client";
 import {
   throwValidationError,
   validateNumber,
   validateObject,
   validateOptionalObject,
   validateString,
 } from "./utils/validation";
+import { toHexString, trimSuffix } from "./utils/string";
+import { SiaSkylink } from "./skylink/sia";
+
+/**
+ * The tus chunk size is 4MiB * dataPieces - encryptionOverhead, set in skyd.
```

Comment:
Skyd's docs say that `chunkSize = (4MiB - encryptionOverhead) * dataPieces`.
That is the same thing if the `encryptionOverhead` is zero, as assumed here.

I'm not sure what the actual value is. For example for `chacha20poly1305` `encryptionOverhead` is 28 bytes.